add a test about boot option descriptions

iximeow opened the pull request:
ensuring we don't inflict [Omicron#5112](https://github.com/oxidecomputer/omicron/issues/5112) on instances due to our own changes: boot option descriptions are load-bearing, so if we've changed them make sure we think through the consequences. i'm pleasantly surprised that `EfiLoadOption` just.. worked here. yay.

the procedure that gets us names we see for guest disks today is a bit subtle, but i've tried to document it fully to pair with the expectation in this test.

there are also specific conditions where failing this assertion is fine - say, if we've done https://github.com/oxidecomputer/propolis/issues/787, or changed how we're processing UEFI boot variables. just, as things are right now, we're surprisingly dependent on dancing with OVMF in exactly the same way every time a guest boots.

gjcolombo reviewed (approved):
This is both a good test for us to have as we (or, well, I) change how NVMe device parameters work and a great description of how our firmware image actually constructs these IDs. Assuming this still passes with the latest HEAD we should merge it!

iximeow commented:
finally coming back to this: yep! gave this a run rebased on current `master` and the test still passes on Alpine, Debian, Ubuntu. ran it with a Windows guest on a lark and realized that this was missing a `phd_skip!()` for Windows guests, so i've added that. once CI's happy, in it oges

iximeow commented:
the `phd-build` failure in https://github.com/oxidecomputer/propolis/commit/392b4b439b0d6e64e4ca4bae655d1d80341876df is because... after i added the `is_linux()` clause to the new test, i rebased that small diff back onto this old branch so i'd have a nice neat tiny diff between what Greg looked at and what i'm gonna merge. `is_linux()`, however, is newer than this test.

so i've just rebased onto master, and it'll _actually_ pass now :sweat:
